src/Surfnet/StepupGateway/SecondFactorOnlyBundle/Service/AuthnContextClassRefValidationService.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+/**
+ * Copyright 2014 SURFnet bv
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Surfnet\StepupGateway\SecondFactorOnlyBundle\Service;
+
+use Psr\Log\LoggerInterface;
+use Surfnet\StepupBundle\Service\LoaResolutionService;
+
+final class AuthnContextClassRefValidationService
+{
+    /**
+     * @var LoggerInterface
+     */
+    private $logger;
+
+    /**
+     * @var LoaAliasLookupService
+     */
+    private $loaAliasLookup;
+
+    /**
+     * @var LoaResolutionService
+     */
+    private $loaResolution;
+
+    public function __construct(
+        LoggerInterface $logger,
+        LoaAliasLookupService $loaAliasLookup,
+        LoaResolutionService $loaResolution
+    ) {
+        $this->logger = $logger;
+        $this->loaAliasLookup = $loaAliasLookup;
+        $this->loaResolution = $loaResolution;
+    }
+
+    /**
+     * @param LoggerInterface $logger
+     * @return $this
+     */
+    public function with(LoggerInterface $logger)
```

Comment:
Actually I extracted it to a separate method because it clutters the signature.

"Resolve this loa id" is cleaner than "Resolve this loa id oh and here is the logger to use".
This way you can pass in a logger with a better context, but it still works fine without.